Modules/posixmodule.c

```diff
@@ -5467,6 +5467,50 @@ os__path_islink_impl(PyObject *module, PyObject *path)
 #endif /* MS_WINDOWS */
 
 
+/*[clinic input]
+os._path_splitroot_ex
+
+    path: unicode
+
+[clinic start generated code]*/
+
+static PyObject *
+os__path_splitroot_ex_impl(PyObject *module, PyObject *path)
+/*[clinic end generated code: output=de97403d3dfebc40 input=f1470e12d899f9ac]*/
+{
+    Py_ssize_t len, drvsize, rootsize;
+    PyObject *drv = NULL, *root = NULL, *tail = NULL, *result = NULL;
+
+    wchar_t *buffer = PyUnicode_AsWideCharString(path, &len);
+    if (!buffer) {
+        goto exit;
+    }
+
+    _Py_skiproot(buffer, len, &drvsize, &rootsize);
+    if (!(drv = PyUnicode_FromWideChar(buffer, drvsize)) ||
+        !(root = PyUnicode_FromWideChar(&buffer[drvsize], rootsize)) ||
+        !(tail = PyUnicode_FromWideChar(&buffer[drvsize + rootsize], len - drvsize - rootsize)))
+    {
+        goto exit;
+    }
+    result = Py_BuildValue("(OOO)", drv, root, tail);
+exit:
+    if (buffer) {
+        PyMem_Free(buffer);
+    }
+    if (drv) {
+        Py_DECREF(drv);
+    }
+    if (root) {
+        Py_DECREF(root);
+    }
+    if (tail) {
+        Py_DECREF(tail);
+    }
```

Comment:
`PyMem_Free` [is a no-op](https://docs.python.org/3/c-api/memory.html#c.PyMem_Free) if called with a `NULL` pointer; `Py_XDECREF` will do the `if (...) Py_DECREF` dance for you.

```suggestion
    PyMem_Free(buffer);
    Py_XDECREF(drv);
    Py_DECREF(root);
    Py_DECREF(tail);
```